paddle/phi/kernels/gpudnn/softmax_gpudnn.h

```diff
@@ -758,11 +761,17 @@ void LaunchNormalSoftmaxBackward(const GPUContext& dev_ctx,
   }
 }
 
-static std::vector<int> GetSoftmaxTensorDims(const phi::DDim& dims,
-                                             const int axis) {
-  int dim = dims[axis];
-  int N = phi::funcs::SizeToAxis(axis, dims);
-  int D = phi::funcs::SizeOutAxis(axis, dims);
+template <typename T = int>
+static std::vector<T> GetSoftmaxTensorDims(const phi::DDim& dims,
+                                           const int axis) {
+  auto dim = static_cast<T>(dims[axis]);
```

Comment:
Why not change the behavior of `SizeToAxis`, such as add a template? Probably some other ops will encounter same problems when calling `SizeToAxis`?

Reply:
Great idea! Would fix soon in next commit.

Reply:
Done.